From fce91df8ba7f305c624baf1f06961c040b088ecc Mon Sep 17 00:00:00 2001
From: Joe Marcus Clarke <marcus@FreeBSD.org>
Date: Mon, 7 Dec 2009 19:05:27 -0500
Subject: [PATCH 1/6] Fix libusb detection on Linux

Adjust the FreeBSD libusb20 code so that it does not interfere with the
Linux libusb detection.

Reported by:	Robby Workman <rw@rlworkman.net>
---
 configure.in |   35 ++++++++++++++++++-----------------
 1 files changed, 18 insertions(+), 17 deletions(-)

Index: hal/configure.in
===================================================================
--- hal.orig/configure.in	2011-02-25 16:27:13.025776431 +0100
+++ hal/configure.in	2011-02-25 16:27:18.565776433 +0100
@@ -482,9 +482,6 @@
 AC_CHECK_HEADERS([linux/videodev.h], [have_videodev_h=yes], [])
 AM_CONDITIONAL(HAVE_V4L1, [test "x$have_videodev_h" = "xyes"])
 
-USE_LIBUSB20=no
-USE_LIBUSB=no
-LIBUSB20_LIBS=""
 AC_ARG_WITH([backend],
 	    AS_HELP_STRING([--with-backend=<name>],
 	                   [backend to use (linux/solaris/freebsd/dummy)]),
@@ -513,21 +510,25 @@
 AM_CONDITIONAL(HALD_COMPILE_SOLARIS, [test x$HALD_BACKEND = xsolaris], [Compiling for Solaris])
 AC_SUBST(HALD_BACKEND)
 if test "x$HALD_BACKEND" = "xfreebsd"; then
-    AC_CHECK_LIB([usb20], [libusb20_dev_get_info], [USE_LIBUSB20=yes], [USE_LIBUSB20=no])
-fi
-if test "x$USE_LIBUSB20" = "xno"; then
-    AC_CHECK_LIB([usb], [libusb20_dev_get_info], [USE_LIBUSB=yes], [USE_LIBUSB=no])
-fi
-AM_CONDITIONAL([HAVE_LIBUSB20],[test "x$USE_LIBUSB20" = "xyes"])
-AM_CONDITIONAL([HAVE_LIBUSB20],[test "x$USE_LIBUSB" = "xyes"])
-if test "x$USE_LIBUSB20" = "xyes"; then
-    AC_DEFINE(HAVE_LIBUSB20, 1, [Set if we need libusb20])
-    LIBUSB20_LIBS="-lusb20"
-elif test "x$USE_LIBUSB" = "xyes"; then
-    AC_DEFINE(HAVE_LIBUSB20, 1, [Set if we need libsub20])
-    LIBUSB20_LIBS="-lusb"
+    USE_BSDLIBUSB20=no
+    USE_BSDLIBUSB=no
+    LIBUSB20_LIBS=""
+    AC_CHECK_LIB([usb20], [libusb20_dev_get_info], [USE_BSDLIBUSB20=yes], [USE_BSDLIBUSB20=no])
+    if test "x$USE_BSDLIBUSB20" = "xno"; then
+        AC_CHECK_LIB([usb], [libusb20_dev_get_info], [USE_BSDLIBUSB=yes], [USE_BSDLIBUSB=no])
+    fi
+    AM_CONDITIONAL([HAVE_LIBUSB20],[test "x$USE_BSDLIBUSB20" = "xyes" -o "x$USE_BSDLIBUSB" = "xyes"])
+    if test "x$USE_BSDLIBUSB20" = "xyes"; then
+        AC_DEFINE(HAVE_LIBUSB20, 1, [Set if we need libusb20])
+        LIBUSB20_LIBS="-lusb20"
+    elif test "x$USE_BSDLIBUSB" = "xyes"; then
+        AC_DEFINE(HAVE_LIBUSB20, 1, [Set if we need libsub20])
+        LIBUSB20_LIBS="-lusb"
+    fi
+    AC_SUBST(LIBUSB20_LIBS)
+else
+    AM_CONDITIONAL([HAVE_LIBUSB20], [false])
 fi
-AC_SUBST(LIBUSB20_LIBS)
 
 dnl DBUS API is subject to changes
 AC_DEFINE_UNQUOTED(DBUS_API_SUBJECT_TO_CHANGE, [], [DBUS API is subject to change])
